apps/antalmanac/src/components/RightPane/CoursePane/SearchForm/AdvancedSearch/AdvancedSearchTextFields.tsx

```
@@ -115,6 +109,7 @@ export function AdvancedSearchTextFields() {
                     break;
                 }
                 default:
+                    console.error(`Could not find advanced search field with name ${name}`);
                     break;
```

issue: ultimately this is still a silent failure

suggestion: add a utility to lib/utils that encodes this in logic

```ts
export const safeUnreachableCase = <T>(
  v: never,
  retVal?: T,
): T => {
  // if this code is running, v didn't turn out to be `never` after all, so tell TS that
  const castedV = v as unknown

  console.error(`Reached a (safe) unreachable case: ${v}`)
  return retVal
}
```

This utility will ensure, when used in the `default` of a switch statement, that the switch value has been exhausted (all potential types were handled)